worker/task/process_event_delivery.go

```diff
@@ -43,8 +45,14 @@ type SignatureValues struct {
 	Timestamp string
 }
 
-func ProcessEventDelivery(appRepo datastore.ApplicationRepository, eventDeliveryRepo datastore.EventDeliveryRepository, groupRepo datastore.GroupRepository, rateLimiter limiter.RateLimiter) func(*queue.Job) error {
+func ProcessEventDelivery(appRepo datastore.ApplicationRepository, eventDeliveryRepo datastore.EventDeliveryRepository, groupRepo datastore.GroupRepository, rateLimiter limiter.RateLimiter, tracer tracer.Tracer) func(*queue.Job) error {
 	return func(job *queue.Job) error {
+		var txn *newrelic.Transaction
+		if tracer != nil {
+			txn = tracer.StartTransaction("ProcessEventDelivery")
+		}
+		defer txn.End()
```

Comment:
Should this guy also be in the if condition?

Reply:
No, if we put it in the condition won't the transaction be scoped to the if condition?

Reply:
Yes it would, because if the condition is false, txn will be `nil` and calling txn.End will panic